### PR TITLE
Secrets header version badge

### PR DESCRIPTION
**Pull request by zofskeez**
Small change to always show the version badge in the secrets engine header.

image

### CODE REVIEW

File: ui/app/templates/components/secret-list-header.hbs [hunk 1]
```diff
@@ -16,11 +16,9 @@
       <h1 class="title is-3">
         <Icon @glyph={{or @model.engineType "secrets"}} @size="xl" class="has-text-grey-light" />
         {{@model.id}}
-        {{#if (eq @model.options.version 2)}}
```
Comment (hashishaw) [inline ui/app/templates/components/secret-list-header.hbs:19]: Only KV type backends should have this badge. This component is used in every type of backend, so we still need some check here to make sure we're only rendering on mount type KV

**Review by hashishaw — CHANGES_REQUESTED** [review]
Might be worth adding a little test too

File: ui/app/templates/components/secret-list-header.hbs (hunk 1 above)
Comment (zofskeez) [reply ui/app/templates/components/secret-list-header.hbs:19]: Thanks for pointing that out @hashishaw. I'll update and add a test. Quick question regarding `engineType`. I see in some places there is check for both `kv` and `generic` types. Is `generic` a pre kv type from older versions that we are now mapping to kv?

Comment (hashishaw) [reply ui/app/templates/components/secret-list-header.hbs:19]: That is a great question. Let me check and get back to you 👀

Comment (hashishaw) [reply ui/app/templates/components/secret-list-header.hbs:19]: Yes, `generic` was "renamed" to `kv` but we keep it for backwards compatibility 👍

**Comment by zofskeez** [comment]
@hashishaw I made the change to only show the badge for `kv` and `generic` engine types and added a test. I also added a mirage factory for the `secret-engine` model as well as a test helper to return ember data models when given models generated by mirage. This could be convenient in integration tests when the component expects a model as an argument rather than stubbing out a pojo.

**Review by hashishaw — APPROVED** [review]
Great work, thank you!

**Pull request merged**